plugin/pkg/kubectl/client/plugin_client.go

```diff
@@ -0,0 +1,192 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"encoding/json"
+	"fmt"
+	"os"
+	"time"
+
+	restclient "k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/kubernetes-incubator/service-catalog/pkg/client/clientset_generated/clientset"
+)
+
+//PluginClient is a client for interacting with the service catalog
+//via generated clientset interface
+type PluginClient struct {
+	ScClient clientset.Interface
+	Config   *restclient.Config
+}
+
+// NewClient uses the KUBECONFIG environment variable to create a new client
+// based on an existing configuration
+func NewClient() (*PluginClient, error) {
+	// resolve kubeconfig location, prioritizing the --config global flag,
+	// then the value of the KUBECONFIG env var (if any), and defaulting
+	// to ~/.kube/config as a last resort.
+	home := os.Getenv("HOME")
+	kubeconfig := home + "/.kube/config"
+
+	kubeconfigEnv := os.Getenv("KUBECONFIG")
+	if len(kubeconfigEnv) > 0 {
+		kubeconfig = kubeconfigEnv
+	}
+
+	configFile := os.Getenv("KUBECTL_PLUGINS_GLOBAL_FLAG_CONFIG")
+	kubeConfigFile := os.Getenv("KUBECTL_PLUGINS_GLOBAL_FLAG_KUBECONFIG")
+	if len(configFile) > 0 {
+		kubeconfig = configFile
+	} else if len(kubeConfigFile) > 0 {
+		kubeconfig = kubeConfigFile
+	}
+
+	if len(kubeconfig) == 0 {
+		return nil, fmt.Errorf("error iniializing client. The KUBECONFIG environment variable must be defined")
+	}
+
+	clientConfig, _, err := clientFromConfig(kubeconfig)
+	if err != nil {
+		return nil, fmt.Errorf("error obtaining client configuration: %v", err)
+	}
+
+	err = applyGlobalOptionsToConfig(clientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error processing global plugin options: %v", err)
+	}
+
+	c, err := clientset.NewForConfig(clientConfig)
+	if err != nil {
+		return nil, fmt.Errorf("error obtaining a client from existing configuration: %v", err)
+	}
+
+	pluginClient := PluginClient{c, clientConfig}
+	return &pluginClient, nil
+}
+
+func clientFromConfig(path string) (*restclient.Config, string, error) {
```

Comment:
All of this is already done in `kubectl`. We should make it reusable (likely part of client-go).